File: 剑指offer全解/64滑动窗口的最大值.py
# -*- coding:utf-8 -*-
class Solution:
    def maxInWindows(self, num, size):
        # write code here
        res = []
        if size < 1 or num is None:
            return res
        n = len(num)-size+1
        for i in range(n):
            temp = max(num[i: i+size])
            res.append(temp)
        return res

File: 剑指offer全解/64滑动窗口的最大值.java
// Brute Force O(n^2) 更简便解法见C++
import java.util.ArrayList;
public class Solution {
    public ArrayList<Integer> maxInWindows(int [] num, int size)
    {
        if(num == null || size < 1) return new ArrayList<Integer>();
        ArrayList<Integer> res = new ArrayList<>();
        for(int i = 0; i+size-1 < num.length; i++) {
            res.add(getMax(num, i, i+size-1));
        }
        return res;
    }
    public int getMax(int[] num, int start, int end) {
        int max = Integer.MIN_VALUE;
        for(int i = start; i <= end; i++) {
            if(num[i] > max) max = num[i];
        }
        return max;
    }
}